partiql-types/src/lib.rs

```diff
@@ -156,76 +170,129 @@ pub enum TypeKind {
     Struct(StructType),
     Bag(BagType),
     Array(ArrayType),
+    // Serves as Bottom Type
+    Unknown,
```

Comment:
The original intention was to have a type as [Bottom Type](https://en.wikipedia.org/wiki/Bottom_type) as the comment suggests. I foresee this coming up in the operator typing in which we end up with an `Unknown` type for one of the operands so that we can call it an _ill-typed_ expression in order to be able to decide what to do next. I don't have a strong opinion for the name and can change.